src/NuGet.Core/NuGet.Commands/RestoreCommand/Diagnostics/UnexpectedDependencyMessages.cs

```diff
@@ -179,7 +179,7 @@ public static IEnumerable<RestoreLogMessage> GetProjectDependenciesMissingLowerB
                    .OrderBy(e => e.Name, StringComparer.OrdinalIgnoreCase)
                    .Select(e => RestoreLogMessage.CreateWarning(
                        code: NuGetLogCode.NU1604,
-                       message: string.Format(CultureInfo.CurrentCulture, Strings.Warning_ProjectDependencyMissingLowerBound,
+                       message: string.Format(CultureInfo.CurrentCulture, Strings.Warning_ProjectDependencyMissing_Version_LowerBound,
```

Comment:
I think it's better if we differentiate between the types of messages, rather than overloading on message. 

So if the version range is null, say one thing, but if it's actually missing a lower bound keep this message (or a variation of it).

Reply:
Ok. I created new NU1609 for null/empty version range.

Reply:
I'm not sure I'm necessarily suggesting a new warning code, but rather just a new message. 

There are pros and cons to new error codes that need to be accounted for. 

* Is the problem distinct enough to warrant a new code? This means additional overhead such as new docs for example.
* People might have this warning suppressed, and now if we switch the error code, that goes away? 
* Different warning codes across different tooling for the same problem.

Reply:
Do some thinking/chat with other team members and @JonDouglas to determine what's more appropriate.